tests/Fixer/Operator/NoUselessConcatOperatorFixerTest.php

```diff
@@ -136,6 +136,20 @@ public static function provideFixCases(): iterable
             ',
         ];
 
+        yield 'do not fix if the execution result would be different' => [
+            '<?php
+                echo "abc $d" . "[$e]";
+                echo "abc $d" . "[3]";
+                echo "abc $d" . \'[3]\';
+                echo "abc $d" . "->e";
+                echo "abc $d" . \'->e\';
+                echo "abc $d" . "->$e";
+                echo "abc $d" . "?->e";
+                echo "abc $d" . "?->$e";
+                echo "abc $d" . \'?->e\';
+            ',
+        ];
+
```

Comment:
Added more tests that fails currently but needs to be passed. b7e764c

Reply:
@Wirone @julienfalque 
Can you think of any other further cases of code breaking due to string concatenation? 🤔

Reply:
Resolved 3b0e1a6